vimiv/imutils/exif.py

```diff
@@ -10,139 +10,269 @@
-def exif_date_time(filename) -> str:
-    """Exif creation date and time of filename."""
-    with contextlib.suppress(piexif.InvalidImageDataError, FileNotFoundError, KeyError):
-        exif_dict = piexif.load(filename)
-        return exif_dict["0th"][piexif.ImageIFD.DateTime].decode()
-    return ""
+        if piexif and check_piexif:
+            return _ExifHandlerPiexif
 
+        if return_value is not None:
+            return return_value
 
-class ExifInformation(dict):
-    """Dictionary to load and store exif information of a single image.
+        return _ExifHandlerNoExif
 
-    Exif information is loaded and formatted upon construction. Afterwards the class
-    behaves like a regular dictionary.
+    return decorator
 
-    Attributes:
-        _exif: Loaded piexif exif information dictionary if any.
-    """
 
-    def __init__(self, filename):
-        super().__init__()
-        self._exif = None
+@check_exif_dependancy()
+class ExifHandler(_ExifHandler):
+    """Main ExifHandler implementation bases on py3exiv2."""
 
-        if piexif is None:
-            log.error("%s relies on exif support", self.__class__.__qualname__)
-        else:
-            self._load_exif(filename)
+    def __init__(self, filename=""):
 
-    def _load_exif(self, filename):
-        """Load exif information from filename into the dictionary."""
         try:
-            self._exif = piexif.load(filename)
-            desired_keys = [
-                e.lower().strip()
-                for e in api.settings.metadata.current_keyset.value.split(",")
-            ]
-            _logger.debug(f"Read metadata.current_keys {desired_keys}")
-        except (piexif.InvalidImageDataError, FileNotFoundError, KeyError):
+            self._metadata = pyexiv2.ImageMetadata(filename)
+            self._metadata.read()
+
+        except FileNotFoundError:
+            _logger.debug("File %s not found", filename)
             return
 
-        for ifd in self._exif:
-            if ifd == "thumbnail":
-                continue
-
-            for tag in self._exif[ifd]:
-                keyname = piexif.TAGS[ifd][tag]["name"]
-                keytype = piexif.TAGS[ifd][tag]["type"]
-                val = self._exif[ifd][tag]
-                _logger.debug(
-                    f"name: {keyname} type: {keytype} value: {val} tag: {tag}"
-                )
-                if keyname.lower() not in desired_keys:
-                    _logger.debug(f"Ignoring key {keyname}")
+    def get_formatted_exif(self) -> Dict[str, str]:
+
+        desired_keys = [
+            e.strip() for e in api.settings.metadata.current_keyset.value.split(",")
+        ]
+        _logger.debug(f"Read metadata.current_keys {desired_keys}")
+
+        exif = dict()
+
+        for key in desired_keys:
+            # For backwards compability, ssume it has one of the following prefixes
+            for prefix in ["", "Exif.Image.", "Exif.Photo."]:
+                key = f"{prefix}{key}"
```

Comment:
We need something like
```
        for base_key in desired_keys:
            # For backwards compability, ssume it has one of the following prefixes
            for prefix in ["", "Exif.Image.", "Exif.Photo."]:
                key = f"{prefix}{base_key}"
```
otherwise we get `Exif.Image.Exif.Photo.Whatever`. That is a nasty little bug that causes the one test to fail as the `ExposureTime` key is not recognized.

Reply:
Oh silly me, good catch!

Reply:
Fixed in 124fb46fbd3343d8209225d4c20c36771a87a764